tests/init.sh

```diff
@@ -1,8 +1,14 @@
+set -e -o pipefail
+
 source common.sh
 
 test -n "$TEST_ROOT"
 if test -d "$TEST_ROOT"; then
     chmod -R u+w "$TEST_ROOT"
+    # We would delete any daemon socket, so let's stop the daemon first.
+    if [[ -n "${NIX_DAEMON_PACKAGE:-}" ]]; then
```

Comment:
This is not the ideal condition for this, but it works for now. Another PR will improve it.